lib/node_modules/@stdlib/math/base/special/sincospi/lib/sincospi.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var isInfinite = require( '@stdlib/math/base/assert/is-infinite' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var sincos = require( '@stdlib/math/base/special/sincos' );
+var floor = require( '@stdlib/math/base/special/floor' );
+var PI = require( '@stdlib/math/constants/float64-pi' );
+var copysign = require( '@stdlib/math/base/special/copysign' );
+
+
+// MAIN //
+
+/**
+* Simultaneously computes the sine and cosine of a number times π.
+*
+* @private
+* @param {(Array|TypedArray|Object)} out - destination array
+* @param {number} x - input value
+* @returns {(Array|TypedArray|Object)} sine and cosine (in radians)
+*
+* @example
+* var v = sincospi( [ 0.0, 0.0 ], 0.0 );
+* // returns [ ~0.0, ~1.0 ]
+*
+* @example
+* var v = sincospi( [ 0.0, 0.0 ], 0.5 );
+* // returns [ ~1.0, ~0.0 ]
+*
+* @example
+* var v = sincospi( [ 0.0, 0.0 ], 0.1 );
+* // returns [ ~0.309, ~0.951 ]
+*
+* @example
+* var v = sincospi( [ 0.0, 0.0 ], NaN );
+* // returns [ NaN, NaN ]
+*/
+function sincospi( out, x ) {
+	var tmp;
+	var ix;
+	var ar;
+	var r;
+
+	if ( isnan( x ) ) {
+		out[0] = NaN;
+		out[1] = NaN;
+		return out;
+	}
+	if ( isInfinite( x ) ) {
+		out[0] = NaN;
+		out[1] = NaN;
```

Comment:
Missing `return`.

Reply:
This would also be a good time to discuss/resolve the discrepancy between `Inf` handing in [sinpi](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/sinpi/lib/sinpi.js#L54) and [cospi](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/@stdlib/math/base/special/cospi/lib/cospi.js#L60)

Reply:
What is the discrepancy?

Reply:
@Planeshifter You remember why we kept a range error in `cospi`?

Reply:
Sorry for the vagueness. Correct me if I'm wrong, but for analogous input, one returns NaN and the other throws a range error.

Reply:
Yes, correct. I am not sure why the difference. This looks to be the remnants of a prior debate which was never resolved.

Reply:
I am firmly in favor of returning `NaN` here. I think this is more of a holdover from this prior debate though, since we have already settled on returning `NaN` instead of throwing range errors for input values outside the allowed range of special functions. `cospi` is the only function left where we throw a `RangeError`, too.

Reply:
Agreed. I will change `cospi`.